homeassistant/components/vegehub/sensor.py

```diff
@@ -0,0 +1,99 @@
+"""Sensor configuration for VegeHub integration."""
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_IP_ADDRESS,
+    CONF_MAC,
+    UnitOfElectricPotential,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, MANUFACTURER, MODEL
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Vegetronix sensors from a config entry."""
+    sensors = []
+    mac_address = config_entry.data[CONF_MAC]
+    ip_addr = config_entry.data[CONF_IP_ADDRESS]
+    num_sensors = config_entry.runtime_data.num_sensors
+    num_actuators = config_entry.runtime_data.num_actuators
+
+    # We add up the number of sensors, plus the number of actuators, then add one
+    # for battery reading, and one because the array is 1 based instead of 0 based.
+    for i in range(num_sensors + num_actuators + 1):  # Add 1 for battery
+        sensor = VegeHubSensor(
+            mac_address=mac_address,
+            slot=i + 1,
+            ip_addr=ip_addr,
+            dev_name=config_entry.data[CONF_HOST],
+        )
```

Comment:
so you are generating a bunch of numbered sensors here, but I'm wondering: What are they actually representing?